enterprise/core-edge/src/main/java/org/neo4j/coreedge/core/EnterpriseCoreEditionModule.java

```diff
@@ -142,10 +143,20 @@ public void registerProcedures( Procedures procedures )
         CopiedStoreRecovery copiedStoreRecovery = new CopiedStoreRecovery( config,
                 platformModule.kernelExtensions.listFactories(), platformModule.pageCache );
 
+        TransactionIdStore offlineTxIdStore;
+        try
+        {
+            offlineTxIdStore = new ReadOnlyTransactionIdStore( platformModule.pageCache, storeDir );
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( e );
+        }
+
         LocalDatabase localDatabase = new LocalDatabase( platformModule.storeDir, copiedStoreRecovery,
-                new StoreFiles( new DefaultFileSystemAbstraction() ),
-                platformModule.dataSourceManager,
-                platformModule.dependencies.provideDependency( TransactionIdStore.class ), databaseHealthSupplier,
+                new StoreFiles( new DefaultFileSystemAbstraction() ), platformModule.dataSourceManager,
+                platformModule.dependencies.provideDependency( TransactionIdStore.class ), () -> offlineTxIdStore,
```

Comment:
This is again just for reasons of symmetry, since the other txIdStore is also brought in using a supplier.